tests/StorageApiTestCase.php

```diff
@@ -719,6 +731,35 @@ function (self $that, array $config = [], $useExistingBranch = false) {
         ];
     }
 
+    protected function getExistingBranchForTestCase(self $that)
+    {
+        $branchName = $this->generateDevBranchNameForDataProvider($that);
+        $devBranch = new \Keboola\StorageApi\DevBranches($that->_client);
+
+        $branches = $devBranch->listBranches();
+        $branch = null;
+        // get branch detail
+        foreach ($branches as $branchItem) {
+            if ($branchItem['name'] === $branchName) {
+                $branch = $branchItem;
+            }
+        }
+        if (!isset($branch)) {
+            $this->fail(sprintf('Reuse existing branch: branch %s not found.', $branchName));
+        }
+
+        return $branch;
+    }
+
+    protected function createDevBranchForTestCase(self $that)
+    {
+        $branchName = $this->generateDevBranchNameForDataProvider($that);
+        $devBranch = new \Keboola\StorageApi\DevBranches($that->_client);
+
+        $this->deleteBranchesByPrefix($devBranch, $branchName);
```

Comment:
Tady je zajímavé, že to smaže všechny existující - v tom původním to tak bylo, ale tady když je to oddělené tak je to zvláštní. Ale fungovalo to předtím, tak proč ne. 

Ale reálně bych čekal, že ten delete by prefix bude existovat jen v setUp a nikde jinde. Tam se to pročistí a pak test jich může nagenerovat víc, aniž by se mazaly. Ale to by byl větší RF.